consensus/hotstuff/votecollector/statemachine.go

```diff
@@ -123,21 +123,50 @@ func (m *VoteCollector) AddVote(vote *model.Vote) error {
 	return nil
 }
 
-// processVote uses compare-and-repeat pattern to process vote with underlying vote processor
+// processVote uses compare-and-repeat pattern to process vote with underlying vote processor.
+// All expected errors are handled internally. Under normal execution only exceptions are
+// propagated to caller.
+//
+// PREREQUISITE: This method should only be called for votes that were successfully added to
+// `votesCache` (identical duplicates are ok). Therefore, we don't have to deal here with
+// equivocation (same replica voting for different blocks) or inconsistent votes (replica
+// emitting votes with inconsistent signatures for the same block), because such votes were
+// already filtered out by the cache.
```

Comment:
Does this statement still hold? To me, it seems to contradict the fact that we still handle the `DoubleVoteError` below? Maybe I am overlooking or misunderstanding some detail 🤔 ...